src/settings/UserLocale.js

```diff
@@ -0,0 +1,214 @@
+      //    العربية / العربية
+      //    الصينية (الصين) / 中文（中国）
+      label: `${intl.formatDisplayName(l, { type: 'language' })} / ${lIntl.formatDisplayName(l, { type: 'language' })}`,
+    };
+  });
+
+  return locales;
+};
+
+
+
+class UserLocale extends React.Component {
+  static manifest = Object.freeze({
+    configId: '',
+    userExists: {
+      type: 'okapi',
+      path: 'users',
+      fetch: false,
+      accumulate: true
+    },
+    localeConfig: {
+      type: 'okapi',
+      path: 'configurations/entries',
+      fetch: false,
+      accumulate: true,
+      PUT: {
+        path: 'configurations/entries/%{configId}',
+      }
+    },
+  });
+
+  static contextType = CalloutContext;
+
+  constructor(props) {
+    super(props);
+
+    this.localesOptions = localesList(props.intl);
+    this.callout = React.createRef();
+  }
+
+  submit = (values, form) => {
+    const { mutator, intl } = this.props;
+    const configEntry = { ...userLocaleConfig };
+
+    const { locale, timezone, currency } = values;
+
+    // GET the user by username in order to get their UUID.
+    // .then(add id to locale-settings)
+    // .then(check whether there is already a config entry)
+    // .then(if yes, PUT locale; if no, POST locale)
+    // .then(send success callout)
+    // .catch(send failure callout)
+    mutator.userExists.GET({ params: { query: `username==${values.username}` } })
+      .then((res) => {
+        configEntry.userId = res.users[0]?.id;
+        if (!configEntry.userId) {
+          throw new Error(intl.formatMessage({ id: 'ui-developer.passwd.error.missingUser' }, { username: values.username }));
+        }
+      })
+      .then(() => {
+        const query = Object.entries(configEntry)
+          .map(([k, v]) => `"${k}"=="${v}"`)
+          .join(' AND ');
+
+        return mutator.localeConfig.GET({ params: { query } });
+      })
+      .then((res) => {
+        configEntry.value = JSON.stringify({ locale, timezone, currency });
+
+        if (res.configs.length) {
+          mutator.configId.replace(res.configs[0].id);
+          return mutator.localeConfig.PUT(configEntry);
+        } else {
+          return mutator.localeConfig.POST(configEntry);
+        }
+      })
+      .then(() => {
+        this.context.sendCallout({
+          type: 'success',
+          message: (<FormattedMessage id="ui-developer.userLocale.success" values={{ username: values.username }} />)
+        });
+        form.restart();
+      })
+      .catch(e => {
+        this.context.sendCallout({
+          type: 'error',
+          message: e.message,
+        });
+      });
+  };
+
+  render() {
+    return (
+      <Pane
+        defaultWidth="fill"
+        paneTitle={<FormattedMessage id="ui-developer.userLocale.setUserLocale" />}
+      >
+        <Form
+          onSubmit={this.submit}
+          render={({ handleSubmit, submitting, pristine }) => (
+            <form onSubmit={handleSubmit}>
+              <Row>
```

Comment:
You technically don't need any of these Row and Col uses since they're all just full 12-column width.